dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageServiceImpl.java

```diff
@@ -319,19 +304,38 @@ public void cleanup(boolean deleteDbRecords, boolean verbose) throws SQLExceptio
         // Aborting will leave the DB objects around, even if the
         // bitstreams are deleted. This is OK; deleting them next
         // time around will be a no-op.
-        catch (SQLException | IOException sqle)
+        catch (SQLException | IOException e)
         {
-            if (verbose)
-            {
-                System.err.println("Error: " + sqle.getMessage());
-            }
-            context.abort();
-            throw sqle;
+            System.err.println("Error: " + e.getMessage());
+            if (null != context)
+                context.abort();
+            throw e;
         } finally {
             if(context != null){
                 context.restoreAuthSystemState();
             }
         }
+        System.out.format("Removed %d deleted bitstreams.%n", removalCounter);
+    }
+
+    private void deleteDbRecords(Context context, Bitstream bitstream, boolean verbose)
```

Comment:
Make this one protected ? So if a bugfix is needed here people can easily override this method.